docs/src/charm/howto/upgrade-minor.md

```diff
@@ -0,0 +1,202 @@
+
+If the `can-upgrade-to` revision is empty, you are done, there's no
+upgrade currently available on this charm channel.
+
+
+### The pre-upgrade-check
+
+Before running an upgrade, we should check that the cluster is 
+steady and ready for upgrade. The charm will perform checks 
+necessary to confirm the cluster is in safe working order before
+upgrading.
+
+```sh
+$ juju run k8s/leader pre-upgrade-check           
+Running operation 3 with 1 task
+  - task 4 on unit-k8s-1
+
+Waiting for task 4...
+$ 
+```
+
+If no error appears, the `pre-upgrade-check` completed successfully
+
+### Refreshing charm applications
+
+#### Control Plane units (k8s)
+
+In order to upgrade the cluster, first the control-plane nodes should be updated
+following running the `pre-upgrade-check`. 
+
+```sh
+$ juju refresh k8s
+$ juju status k8s --watch 5s
+```
+
+The first command instructs the juju controller to use the new charm revision
+on the application's charm channel to upgrade each unit. The charm code
+will be simultaneously replaced on each unit, then the kubernetes snap
+will be updated unit-by-unit finishing with the juju leader unit for the 
+application.
+
+After the `k8s` charm is upgraded, the application `Version` from `juju status`
+will reflect the version of the control-plane nodes making up the cluster.
+
+#### Worker units (k8s-worker)
+
+After updating the control-plane applications, worker nodes may be upgraded
+following running the `pre-upgrade-check`. 
+
+```sh
+$ juju run k8s-worker/leader pre-upgrade-check
+$ juju refresh k8s-worker
+$ juju status k8s-worker --watch 5s
+```
+
+The first command instructs the juju controller to use the new charm revision
+on the application's charm channel to upgrade each unit. The charm code
+will be simultaneously replaced on each unit, then the kubernetes snap
+will be updated unit-by-unit finishing with the juju leader unit for the 
+application.
+
+After the `k8s-worker` charm is upgraded, the application `Version` from `juju status`
+will reflect the version of the worker nodes making up the cluster.
+
+```{note} Repeat for every application using the k8s-worker charm if
+multiple appear in the same model.
+```
+
+
+## Verify an Upgrade
+
+Once an upgrade is complete and units settle, the output from:
+
+<!-- markdownlint-disable -->
+```sh
+juju status
+```
+<!-- markdownlint-restore -->
+... should indicate that all units are active and the correct
+version of **Kubernetes** is running.
+
+It is recommended that you run a [cluster validation][cluster-validation]
+to ensure that the cluster is fully functional.
+
+<!-- LINKS -->
+
+[backup-restore]:     ../../snap/howto/backup-restore.md
+[cluster-validation]: ./validate.md
+[juju-docs]:          https://juju.is/docs/juju/upgrade-models
+[release-notes]:      ../reference/releases.md
+[upgrade-notes]:      ../reference/upgrade-notes.md
+[upstream-notes]:     https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.31.md#deprecation
```

Comment:
it should be updated ever release.  is there a variable for which release this is?

Reply:
There is a {{version}} variable we have set up like {{product}} but I am not sure how it will integrate into a url

Reply:
@nhennigan is there anything i can do to test it?

Reply:
So I tried to use {{version}} in the url but it wasn't working. There does seem to be a work around but it will take a new configuration file and I think that is out of the scope of this PR. I will add it to the backlog to see if it can be automated in the future

Reply:
@nhennigan would you like me to update the link to point to 1.32 changelog?